src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -654,6 +752,114 @@ void Instance::HandleRequestConstraintBasedForecast(HandlerContext & ctx,
+                    constraint.nominalPower.Value() > mDelegate.GetAbsMaxPower())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast nominalPower " ChipLogFormatX64 " out of range [" ChipLogFormatX64 ", " ChipLogFormatX64 "]",
+                                 ChipLogValueX64(constraint.nominalPower.Value()),
+                                 ChipLogValueX64(mDelegate.GetAbsMinPower()),
+                                 ChipLogValueX64(mDelegate.GetAbsMaxPower()));
+
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                    return;
+                }
+
+                if (!constraint.maximumEnergy.HasValue())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast no value for maximumEnergy");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+                    return;
+                }
+            }
+
+            if (HasFeature(Feature::kStateForecastReporting))
+            {
+                if (!constraint.loadControl.HasValue())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast no loadControl");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+                    return;
+                }
+
+                if (constraint.loadControl.Value() < -100 || constraint.loadControl.Value() > 100)
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast bad loadControl %d", constraint.loadControl.HasValue());
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                    return;
+                }
+            }
+        }
+
+        if (iterator.GetStatus() != CHIP_NO_ERROR)
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+            return;
+        }
+    }
+
+    // Check for overlappping elements
+    {
+        auto iterator = constraints.begin();
+        if (iterator.Next())
+        {
+            // Get the first constraint
+            Structs::ConstraintsStruct::DecodableType prevConstraint = iterator.GetValue();
+
+            // Start comparing next vs prev constraints
+            while (iterator.Next())
+            {
+                const Structs::ConstraintsStruct::DecodableType & constraint = iterator.GetValue();
+                if (constraint.startTime < prevConstraint.startTime ||
+                    prevConstraint.startTime + prevConstraint.duration >= constraint.startTime)
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast overlapping constraint times");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                    return;
+                }
+
+                prevConstraint = iterator.GetValue();
```

Comment:
Might be clearer as:
```suggestion
                prevConstraint = constraint;
```